### ISSUE TITLE
Huge Paypal Issue - Double Charge - Error Message.

### ISSUE THREAD

**Issue by PKMTracker** [paypal/paypal-checkout#504]
### Description

Huge Issue - Double Charges - Error Message just started roughly 2 hours ago:

Users get charged when this happens and the error function gets called. Users report issue on all browsers; desktop and mobile.

Earlier I noticed this link was failing as well. 
https://www.paypal.com/webapps/hermes/api/button/funding?country=US

```
buildActions/actions<.payment.execute</<@https://www.paypalobjects.com/web/res/650/27acd48bf7b7b1b1c79c0fa14e769/js/button.js?build=2.0.2_20171013153124935.unx:4206:35

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8075:46

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8082:83

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8071:37

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8082:83

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8082:83

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8071:37

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/</<@https://www.paypalobjects.com/api/checkout.js:8087:33

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8075:46

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8082:83

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/</<@https://www.paypalobjects.com/api/checkout.js:8087:33

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8075:46

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8082:83

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8082:83

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/</<@https://www.paypalobjects.com/api/checkout.js:8087:33

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8075:46

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8077:37

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8082:83

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8082:83

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.reject@https://www.paypalobjects.com/api/checkout.js:8049:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8067:33

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.resolve@https://www.paypalobjects.com/api/checkout.js:8033:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8088:34

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.resolve@https://www.paypalobjects.com/api/checkout.js:8033:17

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch/<@https://www.paypalobjects.com/api/checkout.js:8088:34

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.dispatch@https://www.paypalobjects.com/api/checkout.js:8062:26

./node_modules/zalgo-promise/src/promise.js/ZalgoPromise</ZalgoPromise.prototype.resolve@https://www.paypalobjects.com/api/checkout.js:8033:17

ZalgoPromise/<@https://www.paypalobjects.com/api/checkout.js:8010:42

request/</<@https://www.paypalobjects.com/web/res/650/27acd48bf7b7b1b1c79c0fa14e769/js/button.js?build=2.0.2_20171013153124935.unx:3066:28
```

### Affected browsers

- [x] Chrome
- [x] Safari
- [x] Firefox

**Comment by bluepnume** [comment]
Thanks for the prompt report. We're working on getting a hotfix out asap.

**Comment by bluepnume** [comment]
Can you please clear your cache, make sure `console.log(paypal.version)` shows `4.0.131`, and let me know if you still see the issue? We've added a hotfix and the deploy has gone through.

**Comment by PKMTracker** [comment]
It does. I'll let you know if it happens anymore. Time to do about 50 refunds.. lol.

**Comment by bluepnume** [comment]
I'm extremely sorry about the wasted time on your side.

This issue was caused by a corner-case that slipped through our tests -- before anything else we're going to make sure that test gap is closed, and this never has a chance of regressing again. I also have plans to add some redundant checks to ensure the payment has definitely not gone through before `actions.payment.execute()` is ever rejected, to avoid the possibility of customers attempting to pay twice. I've added https://github.com/paypal/paypal-checkout/issues/505 to track.
